editoast/src/views/projects.rs

```diff
@@ -144,9 +148,22 @@ impl ProjectWithStudyCount {
     )
 )]
 async fn create(
-    State(db_pool): State<DbConnectionPoolV2>,
+    headers: axum::http::HeaderMap,
+    State(AppState {
+        db_pool_v2: db_pool,
+        role_config,
+        ..
+    }): State<AppState>,
     Json(project_create_form): Json<ProjectCreateForm>,
 ) -> Result<Json<ProjectWithStudyCount>> {
+    let mut authorizer = make_authorizer(&headers, role_config.as_ref(), db_pool.clone()).await?;
+    let authorized = authorizer
+        .check_roles([BuiltinRole::OpsWrite].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
```

Comment:
I suppose all of this would disappear with a middleware?

Reply:
I don't think so. `let mut authorizer = make_authorizer(&headers, role_config.as_ref(), db_pool.clone()).await?;` would, but the actual check would not.